src/modules/dashboard/components/ExpenditurePage/Recipient/Recipient.tsx

```diff
@@ -0,0 +1,267 @@
+        <>
+          <FormSection appearance={{ border: 'bottom' }}>
+            <div className={styles.singleUserContainer}>
+              <UserPickerWithSearch
+                data={subscribedUsers}
+                label={MSG.defaultRecipientLabel}
+                name={`recipients[${index}].recipient`}
+                filter={filterUserSelection}
+                renderAvatar={supRenderAvatar}
+                dataTest="paymentRecipientPicker"
+                itemDataTest="paymentRecipientItem"
+                placeholder="Search"
+                sidebarRef={sidebarRef}
+              />
+            </div>
+          </FormSection>
+          <FieldArray
+            name={`recipients[${index}].value`}
+            render={(arrayHelpers) => (
+              <FormSection appearance={{ border: 'bottom' }}>
+                {tokens?.map((token, idx) => (
+                  <div className={styles.valueContainer} key={idx}>
+                    <div className={styles.inputContainer}>
+                      <InputLabel label={MSG.defaultValueLabel} />
+                      <Input
+                        name={`recipients[${index}].value[${idx}].amount`}
+                        appearance={{
+                          theme: 'underlined',
+                          size: 'small',
+                        }}
+                        label={MSG.defaultValueLabel}
+                        placeholder="Not set"
+                        formattingOptions={{
+                          numeral: true,
+                          // @ts-ignore
+                          tailPrefix: true,
+                          numeralDecimalScale: 10,
+                        }}
+                        maxButtonParams={{
+                          setFieldValue,
+                          // mock, needs to be changed to the actual value
+                          maxAmount: '0',
+                          fieldName: `recipients[${index}].value[${idx}].amount`,
+                        }}
+                        elementOnly
+                      />
+                    </div>
+                    <div className={styles.tokenWrapper}>
+                      <div className={styles.removeWrapper}>
+                        {tokens.length > 1 && (
+                          <Button
+                            type="button"
+                            onClick={() => arrayHelpers.remove(idx)}
+                            appearance={{ theme: 'dangerLink' }}
+                          >
+                            <FormattedMessage {...MSG.removeTokenText} />
+                          </Button>
+                        )}
+                      </div>
+                      <div>
+                        <TokenSymbolSelector
+                          label=""
+                          tokens={tokensData}
+                          // eslint-disable-next-line max-len
+                          name={`recipients[${index}].value[${idx}].tokenAddress`}
```

Comment:
Nesting and accessors here are getting out of hand really...

Reply:
There are two nested arrays in this form, the first with recipients, the second with values. That's why names look like that.
<img width="438" alt="Screenshot 2022-06-10 at 12 37 18" src="https://user-images.githubusercontent.com/91876137/173047968-7e60876a-6cda-4fff-ae95-ffe38cc90dd6.png">